crates/biome_css_semantic/src/semantic_model/specificity.rs

```diff
@@ -0,0 +1,180 @@
+                        relative_selector
+                            .map_or(ZERO_SPECIFICITY, |s| evaluate_any_relative_selector(&s))
+                    })
+                    .reduce(|acc, e| acc.max(e))
+                    .unwrap_or(ZERO_SPECIFICITY);
+                base + list_max
+            } else {
+                ZERO_SPECIFICITY
+            }
+        }
+        AnyCssPseudoClass::CssPseudoClassFunctionSelector(s) => {
+            if let Some(base) = s
+                .name()
+                .ok()
+                .and_then(|name| evaluate_pseudo_function_selector(name.text()))
+            {
+                base + s.selector().map_or(ZERO_SPECIFICITY, |selector| {
+                    evaluate_any_selector(&selector)
+                })
+            } else {
+                ZERO_SPECIFICITY
+            }
+        }
+        AnyCssPseudoClass::CssPseudoClassFunctionSelectorList(selector_list) => {
+            if let Some(base) = selector_list
+                .name()
+                .ok()
+                .and_then(|name| evaluate_pseudo_function_selector(name.text()))
+            {
+                let list_max = selector_list
+                    .selectors()
+                    .into_iter()
+                    .map(|selector| {
+                        selector.map_or(ZERO_SPECIFICITY, |s| evaluate_any_selector(&s))
+                    })
+                    .reduce(|acc, e| acc.max(e))
+                    .unwrap_or(ZERO_SPECIFICITY);
+                base + list_max
+            } else {
+                ZERO_SPECIFICITY
+            }
+        }
+        AnyCssPseudoClass::CssPseudoClassFunctionValueList(_) => CLASS_SPECIFICITY,
+        AnyCssPseudoClass::CssPseudoClassIdentifier(_) => CLASS_SPECIFICITY,
+    }
+}
+
+fn evaluate_pseudo_selector(selector: &CssPseudoClassSelector) -> Specificity {
+    match selector.class() {
+        Ok(any_pseudo_class) => evaluate_any_pseudo_class(&any_pseudo_class),
+        Err(_) => ZERO_SPECIFICITY,
+    }
+}
+
+fn evaluate_any_subselector(selector: &AnyCssSubSelector) -> Specificity {
+    match selector {
+        AnyCssSubSelector::CssAttributeSelector(_) => TYPE_SPECIFICITY,
+        AnyCssSubSelector::CssBogusSubSelector(_) => ZERO_SPECIFICITY,
+        AnyCssSubSelector::CssClassSelector(_) => CLASS_SPECIFICITY,
+        AnyCssSubSelector::CssIdSelector(_) => ID_SPECIFICITY,
+        AnyCssSubSelector::CssPseudoClassSelector(s) => evaluate_pseudo_selector(s),
+        AnyCssSubSelector::CssPseudoElementSelector(_) => TYPE_SPECIFICITY,
+    }
+}
+
+pub fn evaluate_compound_selector(selector: &CssCompoundSelector) -> Specificity {
+    let nested_specificity = ZERO_SPECIFICITY; // TODO: Implement this
+
+    let simple_specificity = selector
+        .simple_selector()
+        .map_or(ZERO_SPECIFICITY, |s| evaluate_any_simple_selector(&s));
+    let subselector_specificity = selector
+        .sub_selectors()
+        .into_iter()
```

Comment:
```suggestion
        .iter()
```

Reply:
Fixed it https://github.com/biomejs/biome/pull/4097/commits/91f005cdc25cb42261acd326ed17dc3c58f5a397